colorful/fields.py

```diff
@@ -8,21 +10,49 @@
 
 class RGBColorField(CharField):
     """Field for database models"""
-    widget = widgets.ColorFieldWidget
     default_validators = [RegexValidator(regex=forms.RGB_REGEX)]
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, colors=None, *args, **kwargs):
+        self.colors = colors
         kwargs['max_length'] = 7
         super(RGBColorField, self).__init__(*args, **kwargs)
 
     def formfield(self, **kwargs):
         kwargs.update({
             'form_class': forms.RGBColorField,
-            'widget': self.widget,
+            'widget': widgets.ColorFieldWidget(colors=self.colors),
         })
         return super(RGBColorField, self).formfield(**kwargs)
 
     def deconstruct(self):
         name, path, args, kwargs = super(RGBColorField, self).deconstruct()
+        if self.colors is not None:
+            kwargs['colors'] = self.colors
         del kwargs['max_length']
         return name, path, args, kwargs
+
+    def check(self, **kwargs):
+        errors = super(RGBColorField, self).check(**kwargs)
+        if self.colors is not None:
+            # check colors type
+            if not isinstance(self.colors, (list, tuple)):
+                errors.append(Error(
+                    'colors is not iterable',
+                    hint='Define the colors param as list of strings.',
+                    obj=self,
+                    id='colorful.E001'
+                ))
+            else:
+                # check list values reusing form validator
```

Comment:
I usually add comments when the *Why* of the following code is not obvious and try to keep the code simple enough so I don't have to describe *What* the following code is doing.

I think you can drop both comments since it's pretty obvious what the code is doing here.